go/http/api.go

```diff
@@ -121,6 +124,139 @@ func (this *HttpAPI) ListSnapshotsLogicalVolumes(params martini.Params, r render
 	r.JSON(200, output)
 }
 
+// GetMySQLInfo returns basic MySQL info to be used in pre-seed checks
+func (this *HttpAPI) GetMySQLInfo(params martini.Params, r render.Render, req *http.Request) {
+	if err := this.validateToken(r, req); err != nil {
+		return
+	}
+	output, err := dbagent.GetMySQLInfo()
+	if err != nil {
+		r.JSON(500, &APIResponse{Code: ERROR, Message: err.Error()})
+		return
+	}
+	r.JSON(200, output)
+}
+
+// GetMySQLDatabases returns information about database engines and their sizes
+func (this *HttpAPI) GetMySQLDatabases(params martini.Params, r render.Render, req *http.Request) {
+	if err := this.validateToken(r, req); err != nil {
+		return
+	}
+	output, err := osagent.GetMySQLDatabaseInfo()
+	if err != nil {
+		r.JSON(500, &APIResponse{Code: ERROR, Message: err.Error()})
+		return
+	}
+	r.JSON(200, output)
+}
+
+// StartLocalBackup initiates a process of local backup and returns backup folder
+func (this *HttpAPI) StartLocalBackup(params martini.Params, r render.Render, req *http.Request) {
+	if err := this.validateToken(r, req); err != nil {
+		return
+	}
+	output, err := osagent.StartLocalBackup(params["seedId"], params["seedMethod"], params["databases"])
+	if err != nil {
+		r.JSON(500, &APIResponse{Code: ERROR, Message: err.Error()})
+		return
+	}
+	r.JSON(200, output)
+}
+
+// StartStreamingBackup initiates a process of streaming backup to targetHost
+func (this *HttpAPI) StartStreamingBackup(params martini.Params, r render.Render, req *http.Request) {
+	if err := this.validateToken(r, req); err != nil {
+		return
+	}
+	err := osagent.StartStreamingBackup(params["seedId"], params["targetHost"], params["databases"])
+	if err != nil {
+		r.JSON(500, &APIResponse{Code: ERROR, Message: err.Error()})
+		return
+	}
+	r.JSON(200, err == nil)
+}
+
+// Cleanup remove contents of config.Config.MySQLBackupDir
+func (this *HttpAPI) Cleanup(params martini.Params, r render.Render, req *http.Request) {
```

Comment:
Please rename to `CleanupMySQLBackupDir`

Reply:
Fixed in 3ce673a